src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -0,0 +1,558 @@
+            await fs.ReadAsync(memory, new CancellationToken()).ConfigureAwait(false);
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public async Task CS_Analyzer_NoDiagnostic_ReadAsync_AsMemory()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[fs.Length];
+            await fs.ReadAsync(buffer.AsMemory(), new CancellationToken());
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public async Task CS_Analyzer_NoDiagnostic_NoAwait_SaveAsTask()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    public void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[fs.Length];
+            Task t = fs.ReadAsync(buffer, 0, (int)fs.Length);
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public async Task CS_Analyzer_NoDiagnostic_NoAwait_ReturnMethod()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    public Task M(FileStream fs, byte[] buffer)
+    {
+        return fs.ReadAsync(buffer, 0, (int)fs.Length);
+    }
+}
+            ");
+        }
+
+        [Fact]
+        // VB does not support expression bodies for methods
+        public async Task CS_Analyzer_NoDiagnostic_NoAwait_ExpressionBodyMethod()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    public Task M(FileStream fs, byte[] buffer) => fs.ReadAsync(buffer, 0, (int)fs.Length);
+}
+            ");
+        }
+
+        [Fact]
+        public async Task CS_Analyzer_NoDiagnostic_ConfigureAwait()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[fs.Length];
+            await fs.ReadAsync(buffer, 0, (int)fs.Length).ConfigureAwait(false);
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public async Task CS_Analyzer_NoDiagnostic_ContinueWith_ConfigureAwait()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[fs.Length];
+            await fs.ReadAsync(buffer, 0, (int)fs.Length).ContinueWith(c => {}).ConfigureAwait(false);
```

Comment:
I don't know if we want this one to warn, or not.  @stephentoub ?

Reply:
No, we shouldn't warn here. We definitely can't auto-fix it, and with this rule, at least for now, I think we should only warn on things we can fix with 100% certainty.

Reply:
Another approach would be to ensure that the diagnostic message is clear enough so that the code fix handles a subset of cases flagged by the diagnostic, but the cases not handled by fixer are either extremely rare or can easily be manually fixed by the user. Your stance looks fine if you feel this case falls in neither of these buckets.

Reply:
Thanks, @mavasani.  My concern here is leading folks down a bad path.  ValueTask is really meant just to be awaited, and so if the code is doing anything else, I fear even by warning we're going to encourage people to try things that could be problematic.  Hopefully the ValueTask analyzer would then kick in, but it still makes me nervous.  I'd prefer to stick to just the cases it's easy for us to tackle, namely `await stream.Read/WriteAsync(...){.ConfigureAwait(...)};`.

Reply:
Ah, that makes sense then. Thanks for the details.

Reply:
Yeah, I just didn't remember if ValueTask had an inherent ContinueWith.  Since no, I agree with this being something to not report on.

Reply:
Thanks. I'm not reporting on ContinueWith.